timelock-agent/src/main/java/com/palantir/timelock/paxos/AsyncTimeLockServicesCreator.java

```diff
@@ -84,13 +87,25 @@ public TimeLockServices createTimeLockServices(
                         : JavaSuppliers.compose(NonTransactionalLockService::new, rawLockServiceSupplier),
                 client);
 
+        leadershipCreator.executeWhenLostLeadership(() ->
+                metricsManager.deregisterTaggedMetrics(withTagIsCurrentSuspectedLeader(true)));
+
+        leadershipCreator.executeWhenGainedLeadership(() ->
+                metricsManager.deregisterTaggedMetrics(withTagIsCurrentSuspectedLeader(false)));
+
         return TimeLockServices.create(
                 asyncTimelockService,
                 lockService,
                 asyncOrLegacyTimelockService,
                 asyncTimelockService);
     }
 
+    private static Predicate<MetricName> withTagIsCurrentSuspectedLeader(boolean currentLeader) {
+        return metricName -> metricName.safeTags().containsKey(AtlasDbMetricNames.TAG_CURRENT_SUSPECTED_LEADER)
+                && metricName.safeTags().get(AtlasDbMetricNames.TAG_CURRENT_SUSPECTED_LEADER)
+                        .equals(String.valueOf(currentLeader));
```

Comment:
nit: this is `Optional.ofNullable(metricName.safeTags().get(ADMN.TAG_CURRENT_SUSPECTED_LEADER)).filter(x -> x.equals(String.valueOf(currentLeader)).isPresent();` though I'll leave this as personal preference